File: src/frontend/src/app/features/admin/admin.models.ts
export interface AdminUser {
  id: string;
  email: string;
  displayName: string | null;
  preferredLanguage: 'en' | 'pl' | null;
  createdAtUtc: string;
  updatedAtUtc: string;
  lastLoginAtUtc: string | null;
  roles: string[];
}

export interface UpdateAdminUserRequest {
  email: string;
  displayName: string | null;
  preferredLanguage: 'en' | 'pl' | null;
  roles: string[];
}

export interface AdminUserPage {
  items: AdminUser[];
  page: number;
  pageSize: number;
  totalCount: number;
}

File: src/backend/Kotlet.Application/Admin/AdminModels.cs
namespace Kotlet.Application.Admin;

public sealed record UpdateAdminUserRequest(string Email, string? DisplayName, string? PreferredLanguage, string[]? Roles);

public sealed record AdminUserResponse(Guid Id, string Email, string? DisplayName, string? PreferredLanguage,
    DateTime CreatedAtUtc, DateTime UpdatedAtUtc, DateTime? LastLoginAtUtc, string[] Roles);

public sealed record AdminUserPageResponse(IReadOnlyList<AdminUserResponse> Items, int Page, int PageSize, int TotalCount);

public enum AdminUserOperationStatus { Success, NotFound, Conflict, ValidationFailed }

public sealed record AdminUserOperationResult(
    AdminUserOperationStatus Status,
    AdminUserResponse? User = null,
    Dictionary<string, string[]>? ValidationErrors = null);
